crates/typst/src/text/deco.rs

```diff
@@ -393,15 +412,84 @@ enum DecoLine {
     },
 }
 
-/// Add line decorations to a single run of shaped text.
-pub(crate) fn decorate(
-    frame: &mut Frame,
+/// Generate one line decoration to a frame.
+pub(crate) fn decorate_frame_once(
```

Comment:
Would it maybe be simpler to take an `&mut Frame` here directly? Then, we don't need to return a vector with a `bool` and partition it everywhere.

Reply:
Updated. Now it takes a mutable frame as input.